### PR TITLE
Feature/modify interactions algorithm to create an interactions analysed interaction id log table

### PR DESCRIPTION
**Pull request by dvoong**

### CODE REVIEW

File: app/algorithm/interaction_coi_extraction/steps.py [hunk 1]
```diff
@@ -302,28 +317,48 @@ def _analysed_interaction_chunks():
...
+                    'id',
+                    'place',
+                    'standardized_place',
+                    'action',
+                    'type',
+                    'context',
+                    'negation',
+                ],
+                quote='$',
+            )
```
Comment (dvoong) [inline app/algorithm/interaction_coi_extraction/steps.py]: reraise error

File: app/db/models/internal.py [hunk 2]
```diff
@@ -144,6 +144,14 @@ class InteractionsAnalysed(BaseModel):
     negation = _col(_bool)
 
 
+class InteractionsAnalysedInteractionIdLog(BaseModel):
+
+    __tablename__ = 'interactions_analysed_interaction_id_log'
+    __table_args__ = {'schema': 'algorithm'}
+
+    datahub_interaction_id = _col(UUID(as_uuid=True), primary_key=True)
+
```
Comment (dvoong) [inline app/db/models/internal.py]: add analysed_at

File: tests/algorithm/test_interaction_coi_extraction.py [hunk 3]
```diff
@@ -1,41 +1,153 @@
...
                 'notes': 'Google exports to Brussels',
             }
-        ]
+        ],
+    )
+
+    output = [[('Brussels', 'Belgium', 'exported', 'GPE', ['export'], False)]]
+    add_side_effect(output)
+
+    mock_datetime.datetime.now.return_value = datetime.datetime(2020, 1, 1)
```
Comment (marcuspp) [inline tests/algorithm/test_interaction_coi_extraction.py]: This is fine but I do like https://github.com/spulec/freezegun for testing dates and times

File: app/algorithm/interaction_coi_extraction/steps.py [hunk 4]
```diff
@@ -277,16 +286,22 @@ def _analysed_interaction_chunks():
                 "uploading events"
                 f"{f'{batch_count*batch_size}-{batch_count*batch_size+len(rows)}'}"
             )
+
+            datahub_interaction_ids = []
             for row in rows:
                 id = row[0]
```
Comment (marcuspp) [inline app/algorithm/interaction_coi_extraction/steps.py]: Can this be changed from id as its a builtin python  name

File: tests/algorithm/test_interaction_coi_extraction.py (hunk 3 above)
Comment (dvoong) [reply tests/algorithm/test_interaction_coi_extraction.py]: I'll try out freezegun, next test with datetimes :)

**Pull request merged**